packages/bundler/src/modules/MempoolManager.ts

```diff
@@ -137,6 +137,27 @@ export class MempoolManager {
     }
   }
 
+  private checkMultipleRolesViolation (userOp: UserOperation): void {
+    const knownEntities = this.getKnownEntities()
+    requireCond(
```

Comment:
this rule is reversed: if there is an entity in the mempool that uses this sender - that other entity should be removed (that is, we have "sender first, other entities after" policy)

Reply:
As this check is done on the mempool level, I am not sure this makes sense to "find" the UserOp in mempool that uses `address A` as a "paymaster", remove it, and then add a new UserOp that uses `address A` as a "sender".
Instead, I rejected the UserOp with double-role address in "both directions".